horovod/spark/common/util.py

```diff
@@ -539,6 +543,46 @@ def _train_val_split(df, validation):
     return train_df, val_df, validation_ratio
 
 
+_FILE_AVAILABILITY_WAIT_TIMEOUT_SECS = \
+    int(os.environ.get('FILE_AVAILABILITY_WAIT_TIMEOUT_SECS', '30'))
+
+
+def _wait_file_available(store, url_list):
+    """Waiting about _FILE_AVAILABILITY_WAIT_TIMEOUT_SECS seconds (default 30 seconds) to make sure
+    all files are available for reading. This is useful in some filesystems, such as S3 which only
+    providing eventually consistency.
+    """
+    # Import LocalStore here to avoid circular import
+    from horovod.spark.common.store import LocalStore
+    if isinstance(store, LocalStore):
+        return
+
+    def wait_for_file(path):
+        end_time = time.time() + _FILE_AVAILABILITY_WAIT_TIMEOUT_SECS
+        while time.time() < end_time:
+            if store.exists(path):
+                return True
+            time.sleep(0.1)
+        return False
+
+    pool = ThreadPool(min(len(url_list), 64))
+    try:
+        results = pool.map(wait_for_file, url_list)
+        failed_list = [url for url, result in zip(url_list, results) if not result]
+        if failed_list:
+            raise RuntimeError('Timeout while waiting for all parquet-store files to appear at urls {failed_list},'
+                               'Please check whether these files were saved successfully when materializing dataframe.'
+                               .format(failed_list=','.join(failed_list)))
+    finally:
+        pool.close()
+        pool.join()
+
+
+def get_spark_df_saved_file_list(saved_path):
```

Comment:
can we also have a test for this method?

Reply:
I think this method should be private, so `_get_spark_df_saved_file_list`

Reply:
updated.